### DISCUSSION TITLE
rpm for macOS discontinued?

### DISCUSSION THREAD

**Discussion by wynnw** [Homebrew/discussions#4806] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.13-4-gf1d345a
ORIGIN: https://github.com/Homebrew/brew
HEAD: f1d345a60e2c02f7e5c1b7930601b1921432cfb6
Last commit: 7 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 39ef3e876e703c5beb3333b78ef84d9b6e197910
Core tap last commit: 2 hours ago
Core tap branch: master
Core tap JSON: 25 Sep 19:10 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.5.2-arm64
CLT: 15.0.0.0.1.1694021235
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

It appears over the weekend the rpm formula has been moved to linux only with a 4.19.0 update. I've built a lot of developer workflows around being able to build rpm's on macOS due to a variety of issues, and it's been fantastic and many thanks to those who made that possible. It looks like the rpm maintainers no longer want to do the work of supporting macOS which breaks the workflows that have really benefitted us.

For those of us who were using that and relying on that, is the best course going forward just to fork and use our own private formula based off the 4.18.0 release?

**Comment by SMillerDev** [comment]
> For those of us who were using that and relying on that, is the best course going forward just to fork and use our own private formula based off the 4.18.0 release?

You can use `brew extract`.

> It looks like the rpm maintainers no longer want to do the work of supporting macOS which breaks the workflows that have really benefitted us.

https://github.com/Homebrew/homebrew-core/pull/144033 shows the difficulties the Homebrew team had, maybe you can work with the RPM developers to support macOS again so Homebrew can restore support?

**Reply by wynnw** [reply]
Thanks for the tip, I can't make it work - there's some black magic involved with building those old versions. I support the move as I know that it's going to be mostly impossible to get this working on an ongoing basis with the rpm devs, but it was nice while it lasted. I'll have to revisit my choices and find a new way.

**Reply by ZhongRuoyu** [reply]
See also https://github.com/rpm-software-management/rpm/issues/2222#issuecomment-1693170271 where RPM maintainer indicated that maintaining macOS support is not planned.

**Comment by k0ste** [comment]
How-to install the `rpm` 4.18?
It's enough for me, just need to support .rpm packages on Mac (to see what inside container via `mc`)

**Reply by ZhongRuoyu** [reply]
You may use `brew extract` to extract `rpm` 4.18.1 to your own personal tap and install that. See https://github.com/orgs/Homebrew/discussions/4826#discussioncomment-7194788 for details.

CC @shahpa

**Comment by shahpa** [comment]
I am having a similar difficulty. Can not install `rpm`. Any help would be much appreciated.

**Reply by SMillerDev** [reply]
Please take this to the rpm developers.

**Reply by kishaningithub** [reply]
@shahpa See here in this same thread

https://github.com/Homebrew/discussions/discussions/4806#discussioncomment-7108077
